bcex/examples/candles_strategy.py

```diff
@@ -0,0 +1,427 @@
+        """Determines if the candle is a reversal candle
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            dataframe of candles
+
+        Returns
+        -------
+        is_reversal_candle : bool
+        """
+        if df.candle_color.iloc[-1] is None:
+            return False
+        if self.ignore_none_candles:
+            df = df[~df.candle_color.isna()]
+        if df.candle_color.iloc[-1] == df.candle_color.iloc[-2]:
+            return False
+
+        prior_n_candles = df.iloc[(-1 - self.n_candles_before_reversal) : -1]
+        if len(prior_n_candles.candle_color.unique()) == 1:
+            return True
+        return False
+
+    def order_decision_from_candles(self, df):
+        """Looks at candles computes candle color and if it is a reversal candle and places trade
+
+        Parameters
+        ----------
+        df : pd.DataFrame
+            dataframe of candles
+        """
+        self.exchange.cancel_all_orders()
+        df.loc[df.close > df.open, "candle_color"] = "green"
+        df.loc[df.close < df.open, "candle_color"] = "red"
+        df.loc[df.close == df.open, "candle_color"] = None
+        if self.is_reversal_candle(df):
+            logging.info("REVERSAL CANDLE!!!!")
+            if df.candle_color.iloc[-1] == "green":
+                price = df.high.iloc[-1]
+                balance = self.exchange.get_available_balance(self.symbol.split("-")[1])
+                logging.info(
+                    f"Green Candle after at least {self.n_candles_before_reversal} Red canndles, "
+                    f"BUY at high of prior candle {price}"
+                )
+                self.exchange.place_order(
+                    symbol=self.symbol,
+                    order_type=OrderType.LIMIT,
+                    quantity=(balance * self.balance_fraction) / price,
+                    price=price,
+                    side=OrderSide.BUY,
+                    check_balance=True,
+                )
+            if df.candle_color.iloc[-1] == "red":
+                price = df.low.iloc[-1]
+                balance = self.exchange.get_available_balance(self.symbol.split("-")[0])
+                logging.info(
+                    f"Red Candle after at least {self.n_candles_before_reversal} green canndles, "
+                    f"SELL at high of prior candle {price}"
+                )
+                self.exchange.place_order(
+                    symbol=self.symbol,
+                    order_type=OrderType.LIMIT,
+                    quantity=balance * self.balance_fraction,
+                    price=price,
+                    side=OrderSide.SELL,
+                    check_balance=True,
+                )
+        else:
+            logging.info("Not a reversal candle waiting.....")
+
+
+if __name__ == "__main__":
+    logging.basicConfig(
+        format="%(asctime)s [%(levelname)s] %(message)s", level=logging.INFO
+    )
+    ss = MovingAverageStrategy(
```

Comment:
Not really comfortable with a main which plays actual trades in PROD

Reply:
will change to staging